numpy/lib/_nanfunctions_impl.py

```diff
@@ -1538,7 +1537,12 @@ def nanquantile(
     if a.dtype.kind == "c":
         raise TypeError("a must be an array of real numbers")
 
-    q = np.asanyarray(q)
+    # Use dtype of array if possible (e.g., if q is a python int or float).
+    if isinstance(q, (int, float)) and a.dtype.kind == "f":
+        q = np.asanyarray(q, dtype=np.result_type(a, q))
```

Comment:
As above, just `dtype=a.dtype`? I know I looked at this before, so probably there was a reason for this... If so, let's add it to the comment...

Reply:
Nah, I think it was added for good sport probably.  It might do byte-swaps, etc. but that doesn't matter much.

There may have been a reason: slightly saner user dtype support.  But it seems so niche (and unlikely to affect anything), that I am happy to ignore it for now.